keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,914 @@
+
+#### Story 2
+
+- As a Networking/Telco Engineer, my application is sensitive to uncore cache latency. I want my application to be statically assigned to CPUs that correspond to one single uncore cache so that I can get the maximum throughput from my application. Additionally, I have applications that are also static and of different CPU size requirements. The default Static CPU Manager Policy will result in my applications being assigned to two or more uncore caches. I have a large multi-node Kuberentes cluster. I want a scheduling policy to automatically assign CPUs for the best performance across the fewest amount of uncore cahces based on available CPUs without having to worry about what order/size I deploy my applications so I can ensure I get the best performance possible. 
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+- Feature is enabled by a static policy option flag.  
+- It does not change behavior of non-static policy
+- It preserves the behavior of other static options.
+
+## Design Details
+
+- Add `UnCoreCacheID` in `CPUInfo` and modify CPUTopology struct to also track uncorecaches in `topology.go`
+  
+  ```go
+  // CPUTopology contains details of node cpu, where :
+  // CPU  - logical CPU, cadvisor - thread
+  // Core - physical CPU, cadvisor - Core
+  // Socket - socket, cadvisor - Socket
+  // NUMA Node - NUMA cell, cadvisor - Node
+  // UnCoreCacheID - Split L3 Cache Topology, cadvisor
+  type CPUTopology struct {
+  NumCPUs        int
+  NumCores       int
+  NumUnCoreCache int
+  NumSockets     int
+  NumNUMANodes   int
+  CPUDetails     CPUDetails
+  }
+  type CPUInfo struct {
+    NUMANodeID int
+    SocketID   int
+    CoreID     int
+    UnCoreCacheID  int
+  }
+  ```
+  
+- Get the UnCoreCacheID for each CPU from cadvisor.
+
+  ```go
+  func Discover(machineInfo *cadvisorapi.MachineInfo) (*CPUTopology, error) {
+  CPUDetails[cpu] = CPUInfo{
+    CoreID:     coreID,
+    SocketID:   core.SocketID,
+    NUMANodeID: node.Id,
+    UnCoreCacheID:  core.UncoreCaches[0].Id
+    }
+  }
+    ```
+
+- Setting the `CPUManagerPolicy` to `static` followed by adding a proposed option in `CPUManagerPolicyOptions` called `align-cpus-by-uncorecache` and setting it to `true`.
+
+  ```yaml
+  kind: KubeletConfiguration
+  cpuManagerPolicy: static
+  cpuManagerPolicyOptions:
+    align-cpus-by-uncorecache: true
```

Comment:
is this feature incompatible with other cpuManagerPolicyOptions? if so, we should call this out

Reply:
ah I see that's spoofed out below, can this be mentioned earlier?

Reply:
this might be entirely removed I have it on my todo list if we keep code